spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/GcpCloudSqlAutoConfiguration.java

```diff
@@ -161,7 +162,11 @@ public DataSourceProperties cloudSqlDataSourceProperties(
 				LOGGER.warn("Ignoring provided spring.datasource.url. Overwriting it based on the " +
 						"spring.cloud.gcp.sql.instance-connection-name.");
 			}
-			properties.setUrl(cloudSqlJdbcInfoProvider.getJdbcUrl());
+			String jdbcUrl = cloudSqlJdbcInfoProvider.getJdbcUrl();
+			if (StringUtils.hasText(gcpCloudSqlProperties.getIpTypes())) {
+				jdbcUrl = String.format(jdbcUrl + "&ipTypes=%s", gcpCloudSqlProperties.getIpTypes());
```

Comment:
Have you considered putting this logic into `DefaultCloudSqlJdbcInfoProvider` instead?
That class is specifically designed for creating the JDBC URL string.

Reply:
No. I remember that I looked at patching the `jdbcUrlTemplate` in `DatabaseType`, but I didn't have the `GcpCloudSqlProperties` there. I must have overlooked `DefaultCloudSqlJdbcInfoProvider`. 
Code moved.